libsolidity/codegen/CompilerContext.cpp

```diff
@@ -217,7 +217,8 @@ void CompilerContext::appendInlineAssembly(
 		return true;
 	};
 
-	solAssert(assembly::InlineAssemblyStack().parseAndAssemble(*assembly, *m_asm, identifierAccess), "");
+	solAssert(assembly::InlineAssemblyStack().parseAndAssemble(*assembly, *m_asm, identifierAccess), "Failed to assemble inline assembly block.");
+	setStackOffset(startStackHeight);
```

Comment:
This does not insert POP instructions - we should not modify the stack height here.